### ISSUE TITLE
Wrong path for bower_components after upgrade to v0.4.10

### ISSUE THREAD

**Issue by janteuni** [swarajban/npm-cache#31]
In v0.3.9 it works well, but now the bower_components are in the root of my project and not in the location define by my bowerrc...

![screen shot 2015-12-28 at 13 13 39](https://cloud.githubusercontent.com/assets/6632110/12018605/69310f0c-ad65-11e5-9c1b-699f6def005e.png)

Any idea ?

**Comment by swarajban** [comment]
Hm, that's really strange, I'll look into it this week

**Comment by kstaikov** [comment]
I'm having the same issue.  The directory structure within the tarfile is just 'components/lib1, components/lib2', etc. instead of 'x/y/z/components/file'

**Comment by sod** [comment]
This is fixed in https://github.com/swarajban/npm-cache/pull/39

**Comment by iamsortiz** [comment]
@sod that leads to other problems.

# TL;DR

In my case, i didn't know about this, just installed "yeoman angular-fullstack generator" and I was having problems with grunt serve just after generation because it spected bower_components to be in "client/bower_components" as stated in its .bowerrc
(I :heart: this package, by the way)

Just to consider this policy, ill give you the full example.

# Example, full detail

The example below just expands in more detail what I have already commented

**When running npm-cache install on angular-fullstack yeoman genetator**

```bash
anonymized@anonymized ~/anonymizedFolder/angularfullstack $ npm-cache install
[npm-cache] [INFO] using /home/anonymized/.package_cache as cache directory
[npm-cache] [INFO] [bower] bower_components located at client/bower_components per bowerrc
[npm-cache] [INFO] [bower] config file exists
[npm-cache] [INFO] [bower] cli exists
[npm-cache] [INFO] [bower] hash of /home/anonymized/anonymizedFolder/angularfullstack/bower.json: 197920e9683f8e9473dfe51ad62be702
[npm-cache] [INFO] [bower] cache exists
[npm-cache] [INFO] [bower] clearing installed dependencies at /home/anonymized/anonymizedFolder/angularfullstack/client/bower_components
[npm-cache] [INFO] [bower] ...cleared
[npm-cache] [INFO] [bower] extracting dependencies from /home/anonymized/.package_cache/bower/1.7.7/197920e9683f8e9473dfe51ad62be702.tar.gz
[npm-cache] [INFO] [composer] Dependency config file /home/anonymized/anonymizedFolder/angularfullstack/composer.json does not exist. Skipping install
[npm-cache] [INFO] [npm] config file exists
[npm-cache] [INFO] [npm] cli exists
[npm-cache] [INFO] [npm] hash of /home/anonymized/anonymizedFolder/angularfullstack/package.json: fee6672df27cbc0cd96f2376ac844fec
[npm-cache] [INFO] [npm] cache exists
[npm-cache] [INFO] [npm] clearing installed dependencies at /home/anonymized/anonymizedFolder/angularfullstack/node_modules
[npm-cache] [INFO] [npm] ...cleared
[npm-cache] [INFO] [npm] extracting dependencies from /home/anonymized/.package_cache/npm/3.7.2/fee6672df27cbc0cd96f2376ac844fec.tar.gz
[npm-cache] [INFO] [bower] done extracting
[npm-cache] [INFO] [npm] done extracting
[npm-cache] [INFO] successfully installed all dependencies
```

**As you say, bower_components is at root**

```bash
anonymized@anonymized ~/anonymizedFolder/angularfullstack $ ll
total 144
drwxr-xr-x   8 anonymized anonymized  4096 Feb 24 09:19 .
drwxr-xr-x   4 anonymized anonymized  4096 Feb 23 21:00 ..
drwxr-xr-x  18 anonymized anonymized  4096 Feb 24 09:19 bower_components
-rw-r--r--   1 anonymized anonymized   579 Feb 23 19:25 bower.json
-rw-r--r--   1 anonymized anonymized    47 Feb 23 19:25 .bowerrc
-rw-r--r--   1 anonymized anonymized     0 Feb 23 19:25 .buildignore
drwxr-xr-x   5 anonymized anonymized  4096 Feb 24 09:18 client
drwxr-xr-x   5 anonymized anonymized  4096 Feb 23 19:25 e2e
-rw-r--r--   1 anonymized anonymized   415 Feb 23 19:25 .editorconfig
-rw-r--r--   1 anonymized anonymized   776 Feb 23 19:25 .gitattributes
-rw-r--r--   1 anonymized anonymized   111 Feb 23 19:25 .gitignore
-rw-r--r--   1 anonymized anonymized 20709 Feb 23 19:25 Gruntfile.js
-rw-r--r--   1 anonymized anonymized  1745 Feb 23 19:25 .jscsrc
-rw-r--r--   1 anonymized anonymized  2021 Feb 23 19:25 karma.conf.js
-rw-r--r--   1 anonymized anonymized   390 Feb 23 19:25 mocha.conf.js
drwxr-xr-x 940 anonymized anonymized 36864 Feb 24 09:20 node_modules
-rw-r--r--   1 anonymized anonymized  3401 Feb 23 19:25 package.json
-rw-r--r--   1 anonymized anonymized  2374 Feb 23 19:25 protractor.conf.js
-rw-r--r--   1 anonymized anonymized   975 Feb 23 19:25 README.md
drwxr-xr-x   7 anonymized anonymized  4096 Feb 23 19:25 server
drwxr-xr-x   4 anonymized anonymized  4096 Feb 23 21:02 .tmp
-rw-r--r--   1 anonymized anonymized   190 Feb 23 19:25 .travis.yml
-rw-r--r--   1 anonymized anonymized  1720 Feb 23 19:25 .yo-rc.json
```

**But .bowerrc says**

```bash
anonymized@anonymized ~/anonymizedFolder/angularfullstack $ cat .bowerrc 
{
    "directory": "client/bower_components"
}
```

**So, is missing in it's spected location**

```bash
anonymized@anonymized ~/anonymizedFolder/angularfullstack $ ll client/
total 68
drwxr-xr-x  5 anonymized anonymized  4096 Feb 24 09:18 .
drwxr-xr-x  8 anonymized anonymized  4096 Feb 24 09:19 ..
drwxr-xr-x  5 anonymized anonymized  4096 Feb 23 19:51 app
drwxr-xr-x  3 anonymized anonymized  4096 Feb 23 19:25 assets
drwxr-xr-x 11 anonymized anonymized  4096 Feb 23 19:25 components
-rw-r--r--  1 anonymized anonymized  6774 Feb 23 19:25 favicon.ico
-rw-r--r--  1 anonymized anonymized 24139 Feb 23 19:25 .htaccess
-rw-r--r--  1 anonymized anonymized  4776 Feb 23 21:02 index.html
-rw-r--r--  1 anonymized anonymized   746 Feb 23 19:25 .jshintrc
-rw-r--r--  1 anonymized anonymized    31 Feb 23 19:25 robots.txt
```

**So on "grunt serve" the system crys :ocean:'s of tears in dispair and hopelessness, and also I, because I didn't know what the :fire:h3ll:fire: was going on.**

```bash
   Running "concurrent:server" (concurrent) task

    Running "less:server" (less) task
    >> FileError: '../bower_components/bootstrap/less/bootstrap.less' wasn't found. Tried - client/bower_components/bootstrap/less/bootstrap.less,client/bower_components/bootstrap/less/bootstrap.less,../bower_components/bootstrap/less/bootstrap.less in client/app/app.less on line 1, column 1:
    >> 1 @import '../bower_components/bootstrap/less/bootstrap.less';
    >> 2 @import '../bower_components/bootstrap-social/bootstrap-social.less';
    Warning: Error compiling client/app/app.less Use --force to continue.

    Aborted due to warnings.
```
# Example end

Excuse my long comment.

**Comment by sod** [comment]
have you tried npm-cache 0.6.0? That should extract into the client directory.

**Comment by iamsortiz** [comment]
@sod thank you. **npm-cache 0.6.0 worked like a charm**

I was using 0.5.4 (excuse me, I forgot to mention that before). And even saw the 0.6v was there trough npm-check...but I didn't thougth about to update and try.

Again, thanks.
